### PR TITLE
Enhance RDMA connection handling and interface resolution

### PR DESCRIPTION
**Pull request by moleksy**
- Updated RDMA initialization to use TCP instead of verbs.
- Added function to retrieve network interface name by IP address.
- Improved error handling in RDMA receive thread for connection resets and disconnections.
- Adjusted test cases to reflect new local and remote IP addresses handling.

### CODE REVIEW

File: media-proxy/src/libfabric_dev.c [hunk 1]
```diff
@@ -102,17 +102,17 @@ int rdma_init(libfabric_ctx **ctx)
...
-    hints->mode = FI_CONTEXT;
-    hints->domain_attr->threading = FI_THREAD_SAFE; /* TODO: check performance */
-    hints->addr_format = FI_FORMAT_UNSPEC;
-    hints->ep_attr->type = FI_EP_RDM;
     hints->domain_attr->mr_mode =
         FI_MR_LOCAL | FI_MR_ENDPOINT | (FI_MR_ALLOCATED | FI_MR_PROV_KEY | FI_MR_VIRT_ADDR);
+    hints->ep_attr->type = FI_EP_RDM;
+    hints->caps = FI_MSG;
+    hints->addr_format = FI_SOCKADDR_IN;
+    hints->fabric_attr->prov_name = strdup("tcp");
```
Comment (ko80) [inline media-proxy/src/libfabric_dev.c:110]: Consider using some predefined global const for that like below
```c++
const char * LIB_FABRIC_ATTR_PROV_NAME_TCP = "tcp";
```

File: media-proxy/src/mesh/conn_rdma_rx.cc [hunk 2]
```diff
@@ -142,14 +142,39 @@ void RdmaRx::rdma_cq_thread(context::Context& ctx) {
         } else if (ret == -EAGAIN) {
             // No events to process, yield CPU briefly to avoid busy looping
             std::this_thread::sleep_for(std::chrono::microseconds(CQ_RETRY_DELAY_US));
+        } else if (ret == -FI_EAVAIL) {
+            // Read the error details
+            struct fi_cq_err_entry err_entry;
+            int err_ret = fi_cq_readerr(ep_ctx->cq_ctx.cq, &err_entry, 0);
+            if (err_ret >= 0) {
+                if (err_entry.err == FI_ECONNRESET || err_entry.err == -107) {
+                    log::warn("RDMA connection reset or endpoint not connected. Waiting for new "
```
Comment (ko80) [inline media-proxy/src/mesh/conn_rdma_rx.cc:151]: Don't split the log line like that. Keep it as long as it is. You wouldn't be able to find `Waiting for new connection` in the code.

File: media-proxy/src/mesh/conn_rdma_rx.cc [hunk 3]
```diff
@@ -142,14 +142,39 @@ void RdmaRx::rdma_cq_thread(context::Context& ctx) {
...
+                    thread::Sleep(ctx, std::chrono::milliseconds(1000)); // Pause before retrying
+                } else {
+                    log::error("RDMA rx encountered an error in CQ")(
+                        "error", fi_strerror(err_entry.err))("kind", kind2str(_kind));
+                }
+            } else {
+                log::error("RDMA rx failed to read CQ error entry")
+                          ("error", fi_strerror(-err_ret))("kind", kind2str(_kind));
+            }
+        } else if (ret == -107) {
```
Comment (ko80) [inline media-proxy/src/mesh/conn_rdma_rx.cc:163]: Can this `-107` be changed to some predefined const of the underlying lib that you use?

File: media-proxy/tests/conn_rdma_real_ep_test.cc [hunk 4]
```diff
@@ -139,8 +139,7 @@ class RdmaRealEndpointsTest : public ::testing::Test {
     // Set up RDMA Rx
     mcm_conn_param rx_request = {};
     rx_request.type = is_rx;
-    rx_request.local_addr = {.ip = "192.168.10.21", .port = "8002"};
-    rx_request.remote_addr = {.ip = "192.168.10.20", .port = "8003"};
+    rx_request.local_addr = {.ip = "192.168.1.22", .port = "8002"};
```
Comment (ko80) [inline media-proxy/tests/conn_rdma_real_ep_test.cc:142]: Avoid using `8002` here as it interferes with the legacy TCP listener socket port.

Comment (moleksy) [reply media-proxy/tests/conn_rdma_real_ep_test.cc]: Fixed in https://github.com/OpenVisualCloud/Media-Communications-Mesh/pull/293/commits/907a4e3917a16fa35b4ebde105a15f179531b040

File: media-proxy/src/mesh/conn_rdma_rx.cc (hunk 3 above)
Comment (moleksy) [reply media-proxy/src/mesh/conn_rdma_rx.cc:162]: Fixed in https://github.com/OpenVisualCloud/Media-Communications-Mesh/pull/293/commits/907a4e3917a16fa35b4ebde105a15f179531b040

File: media-proxy/src/mesh/conn_rdma_rx.cc (hunk 2 above)
Comment (moleksy) [reply media-proxy/src/mesh/conn_rdma_rx.cc]: Fixed in https://github.com/OpenVisualCloud/Media-Communications-Mesh/pull/293/commits/907a4e3917a16fa35b4ebde105a15f179531b040

File: media-proxy/src/libfabric_dev.c (hunk 1 above)
Comment (moleksy) [reply media-proxy/src/libfabric_dev.c]: Fixed in https://github.com/OpenVisualCloud/Media-Communications-Mesh/pull/293/commits/907a4e3917a16fa35b4ebde105a15f179531b040

**Review by tszumski — APPROVED** [review]

**Pull request merged**